clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -437,32 +536,78 @@ private void updateGroupSubscription(Set<String> topics) {
 
         assignmentSnapshot = metadataSnapshot;
 
-        log.debug("Finished assignment for group: {}", assignment);
+        log.debug("Finished assignment for group: {}", assignments);
 
         Map<String, ByteBuffer> groupAssignment = new HashMap<>();
-        for (Map.Entry<String, Assignment> assignmentEntry : assignment.entrySet()) {
+        for (Map.Entry<String, Assignment> assignmentEntry : assignments.entrySet()) {
             ByteBuffer buffer = ConsumerProtocol.serializeAssignment(assignmentEntry.getValue());
             groupAssignment.put(assignmentEntry.getKey(), buffer);
         }
 
         return groupAssignment;
     }
 
+    private void adjustAssignment(final Map<TopicPartition, String> ownedPartitions,
```

Comment:
We may have discussed this in the past, but I'm not too thrilled with the consumer mucking around with assignments from the plugin. Couldn't we just raise an error if the assignment doesn't respect ownership?

Reply:
```
Couldn't we just raise an error if the assignment doesn't respect ownership?
```

Hmm, I think that's doable. But on the other hand we are requiring users to be aware how to leverage on `owned-partitions` list and respect it with `COOPERATIVE`. 

I see pros and cons of this approach: not silently mucking around user assignor returned data, but pushing the responsibility of implementing `COOPERATIVE`-supported assignors to users themselves. @ableegoldman wdyt?